src/views/HrReports/RegionReports/RegionReport .jsx

```diff
@@ -1,109 +1,219 @@
-
-import React, { Fragment, useState } from 'react'
-import PageLayoutCloseOnly from 'src/views/CommonCode/PageLayoutCloseOnly'
-import { SELECT_CMP_STYLE } from 'src/views/Constant/Constant'
-import { MdOutlineAddCircleOutline } from 'react-icons/md'
-import { IconButton } from '@material-ui/core'
-import { useHistory } from 'react-router-dom';
+import React, { Fragment, useMemo } from 'react'
+import { useState } from 'react';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useSelector } from 'react-redux'
 import { axioslogin } from 'src/views/Axios/Axios';
-import { warningNofity } from 'src/views/CommonCode/Commonfunc'
-import DistRegionSelect from './DistRegionSelect '
-import EmployeeDistrictSelect from './EmployeeDistrictSelect '
+// import CustomReport from 'src/views/Component/CustomReport'
+import 'ag-grid-community/dist/styles/ag-grid.css'
+import 'ag-grid-community/dist/styles/ag-theme-material.css'
+import { Actiontypes } from 'src/redux/constants/action.type'
+import { ToastContainer } from 'react-toastify'
+import { setDistrict } from 'src/redux/actions/District.Action'
+import Custom_Report_Two from 'src/views/Component/Custom_Report_Two';
+import { setDistRegion } from 'src/redux/actions/DistRegion.Action';
+import { warningNofity } from 'src/views/CommonCode/Commonfunc';
+import RegionSelect from 'src/views/CommonCode/RegionSelect';
 
 const RegionReport = () => {
+    /** Initiliazing values for tabale data*/
+    const [TableData, setTableData] = useState([]);
+    const [value, setValue] = useState(0);
+    const [secondvalue, setSecondvalue] = useState(0)
+    const [secondMenu, setsecondmenu] = useState(0)
+    const [slno, setslno] = useState([])
+    const [text, SetText] = useState(0);
 
-    const history = useHistory()
+    /** To get stored  values from redux */
+    const dispatch = useDispatch();
+    useEffect(() => {
+        dispatch(setDistrict());
+    }, [])
 
-    const [region, setRegion] = useState(0);
-    const [district, setDistrict] = useState(0);
-    const [tableData, setTableData] = useState([]);
-    const [data, setData] = useState(0)
+    /** For district list */
+    const empDistrict = useSelector((state) => {
+        return state.getDistrictList.DidtrictList
+    })
```

Comment:
multiple useSelector used  remove and use once

Reply:
Done